### ISSUE TITLE
bug: SEVERE: hide-ads failed with Twitter

### ISSUE THREAD

**Issue by RobinFrcd** [revanced/revanced-patches#1981]
### Type

Error while patching

### Bug description

Trying to patch twitter 9.71.0 with `revanced-cli-2.20.2-dev.1-all.jar` and `revanced-patches-2.170.0.jar`.

I saw https://github.com/revanced/revanced-patches/issues/1658 but I'm already using dev and latest jars.

### Steps to reproduce

1. Get apk from [APKMirror](https://www.apkmirror.com/apk/twitter-inc/twitter/twitter-9-71-0-release-0-release/twitter-9-71-0-release-0-android-apk-download/)
2. Try to patch with `java -jar revanced-cli-2.20.2-dev.1-all.jar -c -a ~/Downloads/com.twitter.android_9.71.0-release.0-29710000_minAPI21\(arm64-v8a,armeabi-v7a,x86,x86_64\)\(nodpi\)_apkmirror.com.apk -o twitter-revanced-9.71.0.apk -b revanced-patches-2.170.0.jar --keystore=../ks.keystore -m revanced-integrations-0.104.0.apk --exclusive -i dynamic-color -i hide-ads -i hide-recommended-users -i monochrome-icon`

So I'm running latest:
- cli: https://github.com/revanced/revanced-cli/releases/tag/v2.20.2-dev.1
- patches: https://github.com/revanced/revanced-patches/releases/tag/v2.170.0
- integrations: https://github.com/revanced/revanced-integrations/releases/tag/v0.104.0

### Relevant log output

```shell
INFO: Reading dex files 
INFO: Decoding AndroidManifest.xml only, because resources are not needed 
INFO: Skipping change-package-name: Excluded by default 
INFO: Skipping enable-android-debugging: Excluded by default 
INFO: Skipping export-all-activities: Excluded by default 
INFO: Skipping hide-views-stats: Excluded by default 
INFO: Skipping predictive-back-gesture: Excluded by default 
INFO: Skipping remove-screenshot-restriction: Excluded by default 
INFO: Skipping spoof-wifi-connection: Excluded by default 
INFO: Merging revanced-integrations-0.104.0.apk 
INFO: Deleting existing resource cache directory 
INFO: Decoding resources 
INFO: dynamic-color succeeded 
SEVERE: hide-ads failed: 
app.revanced.patcher.patch.PatchResultError: 'hide-ads' depends on 'json-hook' but the following error was raised: Unexpected integrations.
	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:304)
	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:36)
	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:350)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
	at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:83)
	at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:20)
	at app.revanced.cli.command.MainCommand.run(MainCommand.kt:155)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at app.revanced.cli.main.MainKt.main(Main.kt:7)
SEVERE: hide-recommended-users failed: 
app.revanced.patcher.patch.PatchResultError: 'hide-recommended-users' depends on 'json-hook' but the following error was raised: 'json-hook' did not succeed previously
	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:304)
	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:36)
	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:350)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:129)
	at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:83)
	at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:20)
	at app.revanced.cli.command.MainCommand.run(MainCommand.kt:155)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at app.revanced.cli.main.MainKt.main(Main.kt:7)
INFO: Compiling resources 
INFO: Writing modified dex files 
INFO: Aligning com.twitter.android_9.71.0-release.0-29710000_minAPI21(arm64-v8a,armeabi-v7a,x86,x86_64)(nodpi)_apkmirror.com.apk to twitter-revanced-9.71.0_aligned.apk 
INFO: Signing twitter-revanced-9.71.0_aligned.apk to twitter-revanced-9.71.0_signed.apk 
INFO: Found existing keystore: ks.keystore 
INFO: Copying twitter-revanced-9.71.0_signed.apk to twitter-revanced-9.71.0.apk 
INFO: Cleaned up cache directory 
INFO: Finished
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by oSumAtrIX** [comment]
Try using the latest stable CLI

**Comment by RobinFrcd** [comment]
Hi,
Same issue:
`java -jar revanced-cli-2.20.1-all.jar -c -a ~/Downloads/com.twitter.android_9.85.0-release.0-29850000_minAPI23\(arm64-v8a,armeabi-v7a,x86,x86_64\)\(nodpi\)_apkmirror.com.apk -o twitter-revanced-9.85.0.apk -b revanced-patches-2.170.0.jar --keystore=../ks.keystore -m revanced-integrations-0.104.0.apk -e dynamic-color -e hide-recommended-users -e hide-views-stats -e monochrome-icon  --experimental`

```
INFO: Reading dex files 
INFO: Decoding AndroidManifest.xml only, because resources are not needed 
INFO: Skipping change-package-name: Excluded by default 
INFO: Skipping dynamic-color: Manually excluded 
INFO: Skipping enable-android-debugging: Excluded by default 
INFO: Skipping export-all-activities: Excluded by default 
INFO: Skipping hide-recommended-users: Manually excluded 
INFO: Skipping hide-views-stats: Manually excluded 
INFO: Skipping predictive-back-gesture: Excluded by default 
INFO: Skipping remove-screenshot-restriction: Excluded by default 
INFO: Skipping spoof-wifi-connection: Excluded by default 
INFO: Merging revanced-integrations-0.104.0.apk 
SEVERE: hide-ads failed: 
app.revanced.patcher.patch.PatchResultError: 'hide-ads' depends on 'json-hook' but the following error was raised: Unexpected integrations.
	at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:304)
	at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patcher.kt:36)
	at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.kt:350)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:127)
	at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:83)
	at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:20)
	at app.revanced.cli.command.MainCommand.run(MainCommand.kt:155)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at app.revanced.cli.main.MainKt.main(Main.kt:7)
INFO: Not compiling resources because resource patching is not required 
INFO: Writing modified dex files 
INFO: Aligning com.twitter.android_9.85.0-release.0-29850000_minAPI23(arm64-v8a,armeabi-v7a,x86,x86_64)(nodpi)_apkmirror.com.apk to twitter-revanced-9.85.0_aligned.apk 
INFO: Signing twitter-revanced-9.85.0_aligned.apk to twitter-revanced-9.85.0_signed.apk 
INFO: Found existing keystore: ks.keystore 
INFO: Copying twitter-revanced-9.85.0_signed.apk to twitter-revanced-9.85.0.apk 
INFO: Cleaned up cache directory 
INFO: Finished 
```

**Comment by RobinFrcd** [comment]
Same with https://www.apkmirror.com/apk/twitter-inc/twitter/twitter-9-80-0-release-0-release/twitter-9-80-0-release-0-2-android-apk-download/ which was released on March the 14th, the last twitter patch has been updated on March 22.

**Comment by oSumAtrIX** [comment]
Seems like you were using old version of Twitter in OP

**Comment by RobinFrcd** [comment]
I tried all twitter versions, what made it work is downgrading integrations.

**Comment by oSumAtrIX** [comment]
Odd, git diff link?

**Comment by ottosch** [comment]
Worked for me with integrations 0.103.0 too. Did not work with 0.104 to 0.106.

This works:
* Twitter: 9.86.0
* CLI: 2.20.1
* Patches: 2.172.0
* Integrations: 0.103.0

**Comment by kbexit** [comment]
Is this being looked into to be fixed? For us Revanced Manager folks / carrier locked phone owners.

**Comment by Primeey** [comment]
Can confirm this is fixed in [v2.174.0-dev.36](https://github.com/revanced/revanced-patches/releases/tag/v2.174.0-dev.36) and [v0.108.0-dev.22](https://github.com/revanced/revanced-integrations/releases/tag/v0.108.0-dev.22)
